bot/src/discord_commands/jibiki_all_search.js

```diff
@@ -0,0 +1,254 @@
+const {throwPublicErrorInfo} = require('./../common/util/errors.js');
+const {Navigation, NavigationChapter} = require('monochrome-bot');
+const axios = require('axios');
+const constants = require('./../common/constants.js');
+
+const jibikiApiUri = 'https://api.jibiki.app';
+const maxFieldsPerPage = 8;
+
+module.exports = {
+  commandAliases: ['jibiki', 'jb'],
+  cooldown: 3,
+  uniqueId: 'best_dictionary',
+  shortDescription: 'Search Jibiki.',
+  usageExample: '<prefix>jibiki 女らしい',
+  async action(bot, msg, suffix, monochrome) {
+    if (!suffix) {
+      const {prefix} = msg;
+      return throwPublicErrorInfo('jibiki', `Say **${prefix}jibiki [text]** to search for words. For example: **${prefix}jibiki dance**`, 'No suffix');
+    }
+
+    const response = await axios.get(`${jibikiApiUri}/all?query=${encodeURIComponent(suffix)}&sentenceCount=5`);
+
+    if (response.status === 200) {
+      const wordPages = [];
+      const kanjiPages = [];
+      const sentencePages = [];
+
+      for (let page = 0; page < Math.ceil(response.data.length / maxFieldsPerPage); page += 1) {
+        const wordFields = [];
+
+        for (
+          let field = 0;
+          field < response.data.length - (page * maxFieldsPerPage);
+          field += 1
+        ) {
+          const entry = response.data[(page * maxFieldsPerPage) + field];
+          let word = '';
+
+          if (entry.word.forms[0].kanji.info !== null) {
+            word += `*Kanji note; ${entry.word.forms[0].kanji.info}*\n`;
+          }
+          if (entry.word.forms[0].reading.info !== null) {
+            word += `*Reading note; ${entry.word.forms[0].reading.info}*\n`;
+          }
+
+          if (entry.word.jlpt !== null) {
+            word += `**JLPT N${entry.word.jlpt}**\n`;
+          }
+
+          Object.values(entry.word.senses).forEach((sens, i) => {
+            if (sens.part_of_speech.length !== 0) {
+              word += `${i + 1}. (`;
+            }
+            word += sens.part_of_speech.map(pos => pos.short).join(', ');
+            if (sens.part_of_speech.length !== 0) {
+              word += ') ';
+            }
+
+            if (sens.field_of_use.length !== 0) {
+              word += '[';
+            }
+            word += sens.field_of_use.map(fld => fld.long).join(', ');
+            if (sens.field_of_use.length !== 0) {
+              word += '] ';
+            }
+
+            word += `${sens.definitions.join('; ')}\n`;
+            if (sens.miscellaneous.length > 0) {
+              word += `*${sens.miscellaneous.join('; ')}*`;
+            }
+          });
+
+          wordFields.push({
+            name: entry.word.forms[0].kanji.literal === null
+              ? entry.word.forms[0].reading.literal
+              : `${entry.word.forms[0].kanji.literal}【${entry.word.forms[0].reading.literal}】`,
+            value: word,
+          });
+
+          if (entry.kanji.length > 0) {
+            const kanjiFields = [];
```

Comment:
Suggestion: if you want to allow more than one field on the same line, like the Jisho kanji command does, like this:

![image](https://user-images.githubusercontent.com/4397296/73585945-ab5b3d80-4474-11ea-8b02-7b2874f90ce6.png)

You can set `inline` to true on your fields, like `const field = { name: 'name', value: 'value', inline: true };`

Another suggestion/info, if you want to add a kanji or stroke order image to your results, you can add a thumbnail that links an image from my image repo over here: https://github.com/mistval/kanji_images There is some code here that shows how you could do that: https://github.com/mistval/kotoba/blob/master/bot/src/discord/create_kanji_search_page.js#L151